yang/frr-interface.yang

```diff
@@ -24,6 +36,193 @@ module frr-interface {
       "Initial revision.";
   }
 
+  typedef interface-ref {
+    type leafref {
+      path "/frr-interface:lib/frr-interface:interface/frr-interface:name";
+      require-instance false;
+    }
+    description
+      "Reference to an interface";
+  }
+
+  typedef if-flags-type {
+    type enumeration {
+      enum "up" {
+        value 1;
+        description
+          "Interface up.";
+      }
+      enum "broadcast" {
+        value 2;
+        description
+          "Broadcast.";
+      }
+      enum "debug" {
+        value 3;
+        description
+          "Debug type.";
+      }
+      enum "loopback" {
+        value 4;
+        description
+          "Loopback interface.";
+      }
+      enum "pointopoint" {
+        value 5;
+        description
+          "Point-to-Point type.";
+      }
+      enum "notrailers" {
+        value 6;
+        description
+          "";
```

Comment:
Please put a description here

Reply:
Added description to all the interface flags.